components/tikv_util/src/yatp_pool/mod.rs

```diff
@@ -231,15 +231,21 @@ impl<T: PoolTicker> YatpPoolBuilder<T> {
     where
         F: Fn() + Send + Sync + 'static,
     {
-        self.before_stop = Some(Arc::new(f));
+        self.before_stop = Some(Arc::new(move || {
```

Comment:
This is unnecessary. It should be added just like `add_thread_memory_accessor`.

Reply:
You mean we should not change `after_start` and `before_stop` of `YatpPoolBuilder`, but call them with `add_thread_name_to_map` and `remove_thread_name_from_map` every time?

Reply:
I mean calling `add_thread_name_to_map` in the same function which calls `add_thread_memory_accessor`.

Reply:
I put them in the runner of yatp, please take a look again : ).